src/platforms/rp2040/src/lib/networkdriver.c

```diff
@@ -705,6 +810,12 @@ void network_driver_init(GlobalContext *global)
 void network_driver_destroy(GlobalContext *global)
 {
     if (driver_data) {
+        if (driver_data->hostname) {
```

Comment:
if is not required here AFAIK

Reply:
The per interface hostname resides in driver_data until the driver terminates at which point it is freed, but I realize now I wrote this before I fully understood how the driver_destroy functions work. This code should be placed in `network:stop/0,1` which this driver does not yet support.

Internally netif_set_hostname() is defined as:

`#define netif_set_hostname(netif, name) do { if((netif) != NULL) { (netif)->hostname = name; }}while(0)`

If the hostname is freed the interface hostname will immediately revert to `PicoW`, since NULL has the same effect as the hostname being unset.

Reply:
In Pico-SDK examples that set a hostname it is always hardcoded to a string in flash, so there is no additional memory consumed, but to have this set programmatically we don't have that luxury. Perhaps we could come up with an NVS like solution to store this kind of data?

Reply:
Sorry for the confusion. I meant that:

```
if (ptr) {
    free(ptr);
}
```

is equivalent to:

```
free(ptr);
```

as free accepts NULL.